File: modules/common-model/src/main/java/com/shterneregen/securelan/common/model/rtc/RtcSessionMode.java
package com.shterneregen.securelan.common.model.rtc;

public enum RtcSessionMode {
    DATA,
    AUDIO,
    VIDEO,
    AUDIO_VIDEO;

    public boolean audioEnabled() {
        return this == AUDIO || this == AUDIO_VIDEO;
    }

    public boolean videoEnabled() {
        return this == VIDEO || this == AUDIO_VIDEO;
    }
}

File: modules/common-model/src/main/kotlin/com/shterneregen/securelan/common/model/rtc/RtcSessionMode.kt
package com.shterneregen.securelan.common.model.rtc

enum class RtcSessionMode {
    DATA,
    AUDIO,
    VIDEO,
    AUDIO_VIDEO;

    fun audioEnabled(): Boolean = this == AUDIO || this == AUDIO_VIDEO

    fun videoEnabled(): Boolean = this == VIDEO || this == AUDIO_VIDEO
}
